tools/train.py

```diff
@@ -133,6 +214,10 @@ def main():
     model = build_detector(
         cfg.model, train_cfg=cfg.train_cfg, test_cfg=cfg.test_cfg)
 
+    if cfg.data.samples_per_gpu == 'auto':
+        cfg.data.samples_per_gpu = determine_max_batch_size(cfg, distributed)
+        cfg.dump(osp.join(cfg.work_dir, osp.basename(args.config)))
```

Comment:
Why do you need to overwrite the config file?

Reply:
if i don't override the config will contain cfg.data.samples_per_gpu == 'auto', I would like to see actual value, in order to reproduce in future for example. Does it make sense?

Reply:
Yes, it does. But I still have some doubts about the best way of handling this.

Config file becomes different from the one that you actually used in the very beginning. Running the same config with `data.samples_per_gpu == 'auto'` on a different machine with different GPUs will result in setting different batch size. But with a hardcoded value it will lead to a crash.

Also you can always override some config values with an `--update_config` options and config file won't be overwritten with the new values in his case.

So, if you feel that overwriting a config file is the best option here, then Ok. I don't mind.

Reply:
> Also you can always override some config values with an --update_config options and config file won't be overwritten with the new values in his case.
It will be

If I use config for training where samples_per_gpu is defined as 35 and use --update_config samples_per_gpu=auto it will save config with samples_per_gpu=auto, and actual value will be lost.

So I propose to keep it as is.